opencog/atoms/base/Node.cc

```diff
@@ -102,6 +102,14 @@ bool Node::operator<(const Atom& other) const
         return get_type() < other.get_type();
 }
 
+bool Node::operator<(const Value& other) const
+{
+	if (get_type() != other.get_type())
+		return get_type() < other.get_type();
+ 	const Node* nov = (const Node*) &other;
+ 	return get_name() < nov->get_name();
```

Comment:
This would crash is `other` is not a node, right?

This would really need some unit tests, especially since it is apparently used nowhere ATM, so there is no way to evaluate it, even indirectly.

Reply:
Why would it crash if other is not a node? That would mean that the types are different which goes to this linke:  
`return get_type() < other.get_type();` 
Which should work whatever the type of other is.
But it's still a good idea to add some tests for this.

Reply:
You're right, forget about my dumb comment.

Reply:
instead of doing `get_name() < nov->get_name();` you should be comparing hashes. This should be identical to `Node::operator<(const Atom& other)`